src/ownership.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import cockpit from 'cockpit';
+
+import { FileInfo } from './fsinfo';
+
+interface Id {
+    id: number;
+    name: string | null;
+}
+
+interface IdInfo {
+    uid: Id;
+    gid: Id;
+    groups: Id[];
+}
+
+function parse_id(content: string): Id {
+    const match = content.match(/^(\d+)\(([^)]+)\)$/);
+    if (!match) {
+        throw new Error(`invalid id ${content}`);
+    }
+
+    const id = Number(match[1]);
+    const name = match[2] === "unbound" ? null : match[2];
+
+    return { id, name };
+}
+
+export function useId(): IdInfo | null {
+    /* XXX This should be part of the cockpit UserInfo */
```

Comment:
And it is, if you are about it enough, link https://github.com/cockpit-project/cockpit/pull/20517